katello/katello/hostname-change.rb

```diff
@@ -369,18 +381,30 @@ def run
       installer << " --disable-system-checks" if disable_system_check_option?
 
       STDOUT.puts installer
-      installer_output = self.run_cmd("#{installer}")
-      installer_success = $?.success?
-      STDOUT.puts installer_output
-
-      STDOUT.puts "Restarting puppet services"
-      self.run_cmd("/sbin/service puppet restart")
-      self.run_cmd("katello-service restart --only puppetserver")
+      default_installer << ' --disable-system-checks' if disable_system_check_option?
+      failure_message = """
+Something went wrong with the #{@options[:scenario].capitalize} installer.
+Please check the above output and the corresponding logs. Once the issue is resolved
+you may complete the hostname change by running #{default_installer}
+      """
+
+      run_cmd(installer, [0], failure_message) do |result, success|
+        if temp_last_scenario_yaml && temp_last_scenario_yaml.path
+          unless success
+            STDOUT.puts 'restoring last_scenario.yaml'
+            run_cmd("cp #{temp_last_scenario_yaml.path} #{last_scenario_yaml}")
+          end
+          STDOUT.puts 'cleaning up temporary files'
+          temp_last_scenario_yaml.unlink
+        end
 
-      if installer_success
-        self.successful_hostname_change_message
-      else
-        self.fail_with_message("Something went wrong with the #{@options[:scenario].capitalize} installer! Please check the above output and the corresponding logs")
+        if success
+          STDOUT.puts result
+          STDOUT.puts 'Restarting puppet services'
+          run_cmd('systemctl reload-or-try-restart puppet')
+          run_cmd('katello-service restart --only puppetserver')
```

Comment:
Maybe we can do systemctl reload-or-try-restart puppetserver here instead of using katello-service..

Reply:
I was told to use katello-service for any services supported by it, since that will restart them in the way we specify. Is there something puppet-specific that would benefit from `reload-or-try-restart`?